readium/shared/src/main/java/org/readium/r2/shared/publication/Href.kt

```diff
@@ -22,19 +22,18 @@ import timber.log.Timber
 public class Href private constructor(private val href: Url) : Parcelable {
 
     public companion object {
+        /**
+         * Creates an [Href] from a valid URL.
+         */
         public operator fun invoke(href: SharedUrl): Href =
             Href(StaticUrl(href))
 
-        public operator fun invoke(href: String, templated: Boolean = false): Href? {
-            val url =
-                if (templated) {
-                    TemplatedUrl(href)
-                } else {
-                    SharedUrl(href)?.let { StaticUrl(it) }
-                }
-
-            return url?.let { Href(it) }
-        }
+        /**
+         * Creates an [Href] from an URL template (RFC 6570).
+         */
+        @Suppress("RedundantNullableReturnType") // In case we validate the URI template at some point.
+        public fun fromTemplate(href: String): Href? =
+            Href(TemplatedUrl(href))
```

Comment:
What's the rationale? The previous version looks more consistent and convenient to me.

Reply:
The previous API was not convenient to do [the fallback with percent-decoded paths](https://github.com/readium/kotlin-toolkit/pull/412/files#diff-e19d5fe6f81c3453c4c498c325f1c85f2302c159eddc8b902fb4da18509b1e55R203-R219
).

We could keep it but it would be used only in [the LCP `Link` model](https://github.com/readium/kotlin-toolkit/pull/412/files#diff-5e7c723489ee0813620c4ae4548960f4f057aab3b9085b11924d4b2c5e003937R41-R47). IMHO it's a bit weird to have both the separate constructors + a convenience constructor which just relocate the condition.

Reply:
It used to be largely used on literals in tests.

I'm not sure what you would find a bit weird, but I can propose the following API which should fit every need:
- `invoke(href: String, templated: Boolean = false`)
- `invoke(url: Url)`